api/sql.ts

```diff
@@ -28,35 +28,183 @@ async function execute(cmd: string, placeholder?: string[]): Promise<any> {
+ */
+async function addProfGraphQL({
+  profName,
+  firstName,
+  lastName,
+  avgDifficulty,
+  avgRating,
+  numRatings,
+  wouldTakeAgainPercent,
+  id,
+  legacyId,
+}: Professor): Promise<void> {
+  try {
+    // Check if data already exists in db
+    const result = await profSearch(profName);
+    if (!result || Object.keys(result).length === 0) {
+      void execute(
+        `INSERT INTO rateMyProfessorDB (
+        profName, 
+        firstName, 
+        lastName, 
+        avgDifficulty, 
+        avgRating, 
+        numRatings, 
+        wouldTakeAgainPercent,
+        id,
+        legacyId) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)`,
+        [
+          profName.toLowerCase(),
+          firstName,
+          lastName,
+          avgDifficulty.toFixed(1),
+          avgRating.toFixed(1),
+          numRatings.toString(),
+          wouldTakeAgainPercent.toFixed(2),
+          id,
+          legacyId.toString(),
+        ]
+      );
+      console.log(
+        `[SUCCESS] Professor ${profName} has been added to the database.`
+      );
+    } else {
+      console.error(`Professor ${profName} already exists in the database.`);
+    }
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+/**
+ * Updates existing professor data in the SQL database.
+ * @param {Professor} newProfessor See professor interface (/api/Professor.d.ts).
+ */
+export async function updateProf({
+  profName,
+  avgDifficulty,
+  avgRating,
+  numRatings,
+  wouldTakeAgainPercent,
+}: ProfessorUpdate): Promise<void> {
+  try {
+    void execute(
+      `UPDATE rateMyProfessorDB SET
+              avgDifficulty = ?,
+              avgRating = ?,
+              numRatings = ?,
+              wouldTakeAgainPercent = ?,
+              timeAdded = CURRENT_TIMESTAMP
+              WHERE profName = ?`,
+      [
+        avgDifficulty.toFixed(1),
+        avgRating.toFixed(1),
+        numRatings.toString(),
+        wouldTakeAgainPercent.toFixed(2),
+        profName,
+      ]
+    );
+    console.log(`[SUCCESS] Professor ${profName} has been updated.`);
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+/**
+ * Checks if a professor's data in the db is more than 3 months old.
+ * Current expiration date is 3 months, or 7884000 seconds.
+ *
+ * @param {string}  broncoDirectName BroncoDirect name.
+ * @return {Promise<boolean>} True if prof data is 3mo+ old, false otherwise.
  */
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-async function profSearch(broncoDirectName: string): Promise<void> {
+export async function checkExpiredProfData(
+  broncoDirectName: string
+): Promise<boolean> {
+  const name = broncoDirectName.toLowerCase();
   const result = await execute(
-    'SELECT * FROM `rateMyProfessorDB` WHERE `broncoDirectName` = ?',
+    'SELECT * FROM `rateMyProfessorDB` WHERE `profName` = ? AND TIMESTAMPDIFF(SECOND, `timeAdded`, NOW()) >= 7884000',
+    [name]
+  );
+  if (Object.keys(result).length === 0) {
```

Comment:
Return the value of the boolean expression instead of explicitly returning the intended value.
`return Object.keys(result).length !== 0`

Reply:
Fixed, ty